tensorboard/compat/proto/update.sh

```diff
@@ -41,4 +41,57 @@ find tensorboard/compat/proto/ -type f  -name '*.proto' -exec perl -pi \
   -e 's|tensorflow\.TensorShapeProto|tensorboard.TensorShapeProto|g;' \
   {} +
 
+# Update dependency graph.
+(
+  cd tensorboard/compat/proto/
+
+  {
+    # Keep all non-`tb_proto_library` paragraphs from the build file...
+    awk '
+      BEGIN { RS = ""; ORS = "\n\n" }
+      !/tb_proto_library\(/ && !/AUTOMATICALLY GENERATED/
+    ' BUILD
+    printf '# Rest of file AUTOMATICALLY GENERATED by tensorboard/compat/proto/update.sh.\n\n'
+
+    # ...then regenerate the individual proto targets...
+    for f in *.proto; do
+      printf 'tb_proto_library(\n'
+      printf '\tname = "%s",\n' "${f%.proto}"
+      printf '\tsrcs = ["%s"],\n' "$f"
+      if grep -q '^import "tensorboard/' "$f"; then
+        printf '\tdeps = [\n'
+        grep '^import "tensorboard/' "$f" | sort |
+          sed -e 's#.*compat/proto/\([^.]*\).*#\t\t":\1",#'
+        printf '\t],\n'
+      fi
+      printf ')\n\n'
+    done
+
+    # ...as well as `protos_all`.
+    printf '%s\n' \
+      '# Protobuf files copied from the main TensorFlow repository.' \
+      '# Keep this list synced with proto_test.py' \
+      ;
+    printf 'tb_proto_library(\n'
+    printf '\tname = "protos_all",\n'
+    printf '\tsrcs = [],\n'
+    printf '\tvisibility = ["//visibility:public"],\n'
+    printf '\tdeps = [\n'
+    for f in *.proto; do
+      printf '\t\t":%s",\n' "${f%.proto}"
+    done | sort
+    printf '\t],\n'
+    printf ')\n'
+  } | expand -t4 >BUILD.new
```

Comment:
Optional, but I'd rather just use 4 spaces per tab inline.  It's simpler, more like the final output, and IMO it reads better to not have the lines start with `\t` which makes it harder for me to parse the word (i.e. I see `tname` and `tdeps`).

The one argument I could see maybe is the places where we have 8 spaces and it's not visually aligned, but in that case maybe just pipe those individual lines to `expand`?  That way the readability overall doesn't suffer.

Reply:
Hmm… okay. It’s easier for me to count `\t` than `␣␣␣␣`, but sure; done.